### PR TITLE
Gradient rewrite for both Linear and Radial Gradients.

### PR DESCRIPTION
**Pull request by stevesheldon**
Characters looks pretty good to me now.

@articulate/mac - please review.

### CODE REVIEW

**Comment by tchamblee** [comment]
@stevesheldon Is there a Storyline branch pointed at this submodule branch, where this can be tested?

File: source/IJSVGLinearGradient.m [hunk 1]
```diff
@@ -62,60 +62,94 @@ + (NSGradient *)parseGradient:(NSXMLElement *)element
...
-    if(self.units == IJSVGUnitUserSpaceOnUse) {
-        parentXOffset = (rect.origin.x/rect.size.width);
-        parentYOffset = (rect.origin.y/rect.size.height);
+
+    /** 
+     
+     In order to draw correctly, we must follow the procedures in this document
+
+
+     https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/gradientUnits
```
Comment (tchamblee) [inline source/IJSVGLinearGradient.m:74]: Nice comments Steve

**Comment by tchamblee** [comment]
I haven't taken a look at this consumed in Storyline, but the code here LGTM.  Great job referencing the specs and rewriting some of the more gritty drawing code in here.  Impressive work!

👍

**Comment by stevesheldon** [comment]
Not yet because Robert said the reference will never be right until this is merged

**Comment by tchamblee** [comment]
What Robert was saying is that you need to merge the submodule PR first, update the reference to the submodule in your feature branch, and then merge the PR in the feature branch.  Else, the submodule will become outdated every time.

Without a feature branch which points at your changes in the submodule, we have no way of developer testing submodule changes to verify.  Sometimes a review of the code itself isn't sufficient.

**Comment by tchamblee** [comment]
I disagree with not opening the feature branch PR until after the submodule PR is reviewed and merged.  This doesn't allow for developer testing, and it's often very helpful to see change sets of both repositories in context.

**Comment by stevesheldon** [comment]
I get it, in this case the only change on the storyline side is the sub module update.

**Pull request merged**